src/state_machine.zig

```diff
@@ -505,6 +555,97 @@ pub fn StateMachineType(
             self.prefetch_finish();
         }
 
+        fn scan_account_transfers(self: *StateMachine, filter: GetAccountTransfers) void {
+            assert(self.scan_result_count == 0);
+
+            const scan_groove: *TransfersGroove.ScanGroove = &self.forest.grooves.transfers.scan;
+
+            var scan = scan: {
+                const direction: Direction = if (filter.flags.order_by_descending)
+                    .descending
+                else
+                    .ascending;
+
+                // Getting the timestamp range according to the direction:
+                const timestamp_range = if (filter.timestamp == 0)
+                    TimestampRange.all()
+                else switch (direction) {
+                    // The `timestamp` filter is an exclusive range,
+                    // so the user can use the last seen timestamp to get the next block of results:
+                    .ascending => TimestampRange.gte(filter.timestamp + 1),
+                    .descending => TimestampRange.lte(filter.timestamp - 1),
+                };
+
+                // This query may have 2 conditions:
+                // `WHERE debit_account_id = $account_id OR credit_account_id = $account_id`.
+                var scan_conditions: stdx.BoundedArray(*TransfersGroove.ScanGroove.Scan, 2) = .{};
+
+                // If no flag was set, we default to include both credits and debits.
+                const include_all = filter.flags.debits == false and filter.flags.credits == false;
```

Comment:
With ergonomics in mind, I considered calling them `credits_exclude` and `debits_exclude`, so the flags with the default value `0` would include both. Since we can't enforce the default `true` on the client side, I decided to implement a special case. But you're right, it's better to be explicit.

We can return empty results for incorrect parameters since we don't have "space" to return proper errors. The same applies if the user calls `get_account_transfers` with more than a single event (maybe we should design the Query API considering it).

Reply:
> We can return empty results for incorrect parameters since we don't have "space" to return proper errors. The same applies if the user calls get_account_transfers with more than a single event (maybe we should design the Query API considering it).

:+1: 

> But you're right, it's better to be explicit.

I wonder if we should go even farther, and not even have default values at the language client level? i.e. it would be a client error (e.g. thrown exception) if either `debits` or `credits` is omitted. (Maybe for the sort order too: https://github.com/tigerbeetle/tigerbeetle/pull/1054#discussion_r1413994921).